### ISSUE TITLE
[SCREENREADER, KEYBOARD]: 'Close this modal' reads before knowing the context of what the modal

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#5871]
## Description

When a modal is opened, the first thing that is announced is “Close this modal”, then it reads the heading. So, a person would have to tab backwards to close the modal or close it without knowing what the content is. It would be an improved experience if the context of the modal was communicated with the opportunity to close it.

This was an output from the accessibility audit for https://github.com/department-of-veterans-affairs/va.gov-team/issues/5323.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to know what the content of the modal is before I might close the modal.

## Steps to Recreate

1. Enter `https://staging.va.gov/profile` in browser
2. Start screenreading device of your choice
3. Navigate to the edit the Mailing address
4. Verify the content reading order is 
  a. "Close this modal"
  b. Heading

## Possible Fixes (optional)

- If possible, add {title} to the aria-label like “Close this {title} modal”. (Likely the simplest/preferred solution)
- Another option would be to change the order of the elements, so the heading reads before the close button.

**Comment by alexpappasoddball** [comment]
@brandonrapp Throwing your way for prioritization

**Comment by 1Copenut** [comment]
I'm adding Carol as an assignee, because this seems like a design system change that I can't make solo.

**Comment by k80bowman** [comment]
We need to validate if this is still an issue, and if it is, consider the possible fixes above and determine the best course of action to resolve it.

**Comment by k80bowman** [comment]
For reference, this is the current design system modal component: https://design.va.gov/storybook/?path=/docs/components-modal--default

**Comment by CourtneyChambers77** [comment]
When a modal dialog is activated, focus must be moved to the dialog. Where focus is initially placed may vary depending on the dialog’s content but focusing the dialog itself can provide a consistently predictable user experience.
A modal dialog should have an accessible name, announce itself as a dialog, and should provide standard methods for the user to close it. By a close button, by use of esc key, mouse clicking or tapping outside of the dialog, and ensuring F6 will continue to allow the user to move keyboard focus to the browser’s address bar.
While the modal dialog is active, the contents obscured by the modal dialog should be inaccessible to all users. This means that the TAB key, and a screen reader’s virtual cursor (arrow keys) should not be allowed to leave the modal dialog and traverse the content outside of the dialog.
When a modal dialog is closed, focus should return to the control that initially activated the dialog. This will allow keyboard and screen reader users to continue to parse the document from where they left off. If a modal dialog was not initiated by a purposeful user action (boo), or the element that had activated the dialog is no longer in the DOM, then closing the dialog should place the user’s focus in a logical location. If a dialog was opened on page load, then focus could be placed on either the body or main element. If the trigger was removed from the DOM, then placing focus as close to the trigger’s DOM location would be ideal.

**Comment by k80bowman** [comment]
@CourtneyChambers77 would you mind checking our [current modal component](https://design.va.gov/storybook/?path=/docs/components-modal--default) to see if the close button is announced first?

**Comment by CourtneyChambers77** [comment]
the close button is announced when using NVDA

**Comment by k80bowman** [comment]
Thank you @CourtneyChambers77! Moving this card back our Reviewed/Needs Prioritized column so we can prioritize it for work.
